src/OrleansAWSUtils/Reminders/DynamoDBReminderTable.cs

```diff
@@ -0,0 +1,294 @@
+            return storage.InitializeTable(TABLE_NAME_DEFAULT_VALUE,
+                new List<KeySchemaElement>
+                {
+                    new KeySchemaElement { AttributeName = REMINDER_ID_PROPERTY_NAME, KeyType = KeyType.HASH },
+                    new KeySchemaElement { AttributeName = GRAIN_HASH_PROPERTY_NAME, KeyType = KeyType.RANGE }
+                },
+                new List<AttributeDefinition>
+                {
+                    new AttributeDefinition { AttributeName = REMINDER_ID_PROPERTY_NAME, AttributeType = ScalarAttributeType.S },
+                    new AttributeDefinition { AttributeName = GRAIN_HASH_PROPERTY_NAME, AttributeType = ScalarAttributeType.N }
+                });
+        }
+
+        public async Task<ReminderEntry> ReadRow(GrainReference grainRef, string reminderName)
+        {
+            var reminderId = ConstructReminderId(serviceId, grainRef, reminderName);
+
+            var keys = new Dictionary<string, AttributeValue>
+                {
+                    { $"{REMINDER_ID_PROPERTY_NAME}", new AttributeValue(reminderId) },
+                    { $"{GRAIN_HASH_PROPERTY_NAME}", new AttributeValue { N = grainRef.GetUniformHashCode().ToString() } }
+                };
+
+            try
+            {
+                return await storage.ReadSingleEntryAsync(TABLE_NAME_DEFAULT_VALUE, keys,
+                    item => new ReminderEntry
+                    {
+                        ETag = item[ETAG_PROPERTY_NAME].N,
+                        GrainRef = GrainReference.FromKeyString(item[GRAIN_REFERENCE_PROPERTY_NAME].S),
+                        Period = TimeSpan.Parse(item[PERIOD_PROPERTY_NAME].S),
+                        ReminderName = item[REMINDER_NAME_PROPERTY_NAME].S,
+                        StartAt = DateTime.Parse(item[START_TIME_PROPERTY_NAME].S)
+                    }).ConfigureAwait(false);
+            }
+            catch (Exception exc)
+            {
+                logger.Warn(ErrorCode.ReminderServiceBase,
+                    $"Intermediate error reading reminder entry {Utils.DictionaryToString(keys)} from table {TABLE_NAME_DEFAULT_VALUE}.", exc);
+                throw;
+            }
+        }
+
+        public async Task<ReminderTableData> ReadRows(GrainReference grainRef)
+        {
+            var expressionValues = new Dictionary<string, AttributeValue>
+                {
+                    { $":{SERVICE_ID_PROPERTY_NAME}", new AttributeValue(serviceId.ToString()) },
+                    { $":{GRAIN_REFERENCE_PROPERTY_NAME}", new AttributeValue(grainRef.ToKeyString()) }
+                };
+
+            try
+            {
+                var expression = $"{SERVICE_ID_PROPERTY_NAME} = :{SERVICE_ID_PROPERTY_NAME} AND {GRAIN_REFERENCE_PROPERTY_NAME} = :{GRAIN_REFERENCE_PROPERTY_NAME}";
+                var records = await storage.ScanAsync(TABLE_NAME_DEFAULT_VALUE, expressionValues, expression,
+                    item => new ReminderEntry
```

Comment:
wouldn't you want to put that translation code into a func? Its repeated 3 times here.